app/controllers/users/sessions_controller.rb

```diff
@@ -86,4 +91,29 @@ def mock_login
     sign_in(user, scope: :user)
     redirect_to profile_dashboard_path(user)
   end
+
+  def validate_email
+    email = params.dig(:user, :email)
+
+    if email.blank?
+      user = User.new
+      user.errors.add :email, :blank
+      return user
+    end
+
+    unless email.match(/^\S+@\S+\.\S+$/)
+      user = User.new
+      user.errors.add :email, :invalid
+      return user
+    end
+
+    user = User.find_by(email: params[:user][:email])
+
+    if user.blank?
+      user = User.new(email: email)
+      user
+    end
+
+    user
+  end
```

Comment:
There is a rather more sophisticated email format checker in our code already. `NotifyEmailValidator`.

You use it by making a model, and in that model, writing something like
```
  validates :email, notify_email: true
```

I wonder if we want to make a form object for sign in to wire it up nicely.

Reply:
We're using `User` here, which already has ` validates :email, presence: true, uniqueness: true, notify_email: true`, so we probably hijack that

Reply:
Although maybe not, since that would cause issues with uniqueness

Reply:
Oh, interesting, that would be even nicer :)